### ISSUE TITLE
Broken "buffalo test" command in case of missing schema.sql and development database

### ISSUE THREAD

**Issue by ghaabor** [gobuffalo/buffalo#1191]
## Steps to Reproduce the Problem

Please describe in painful detail what you did (so others can play along with you) to get to this point. This includes things like the exact command(s) you used, or the `curl` command you used, that sort of thing.

1. Delete the schema file from the migrations directory
2. Drop all databases 
3. Run `buffalo test`

## Expected Behavior

`buffalo test` should create the schema file and load it into the your-app_test database and tests should start running.

## Actual Behavior

`buffalo test` breaks with `ERRO[0001] Error: unable to load schema for my-app_test: pq: relation "a_database_table" already exists`

## Info

Please run `buffalo info` and paste the information below where it says "PASTE_HERE".

<details>

```
### Buffalo Version
v0.12.3

### App Information
Pwd=/go/src/github.com/bitrise-team/den/server
Root=/go/src/github.com/bitrise-team/den/server
GoPath=/go
Name=server
Bin=bin/server
PackagePkg=github.com/bitrise-team/den/server
ActionsPkg=github.com/bitrise-team/den/server/actions
ModelsPkg=github.com/bitrise-team/den/server/models
GriftsPkg=github.com/bitrise-team/den/server/grifts
VCS=
WithPop=true
WithSQLite=true
WithDep=false
WithWebpack=false
WithYarn=false
WithDocker=false
WithGrifts=true

### Go Version
go version go1.10.3 linux/amd64

### Go Env
GOARCH="amd64"
GOBIN=""
GOCACHE="/root/.cache/go-build"
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/go"
GORACE=""
GOROOT="/usr/local/go"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
GCCGO="gccgo"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build426265726=/tmp/go-build -gno-record-gcc-switches"

### Node Version
v8.11.3

### NPM Version
5.6.0

### Yarn Version
1.7.0

### PostgreSQL Version
PostgreSQL Not Found

### MySQL Version
mysql  Ver 15.1 Distrib 10.1.26-MariaDB, for debian-linux-gnu (x86_64) using readline 5.2

### SQLite Version
3.16.2 2017-01-06 16:32:41 a65a62893ca8319e89e48b8a38cf8a59c69a8209

### Dep Version
could not find a Gopkg.toml file

### Dep Status
could not find a Gopkg.toml file

```

</details>

**Comment by ghaabor** [comment]
This issue wasn't present in 0.11.x. Traced it back, it's caused by the new FileMigrator logic in the `buffalo test` command.

https://github.com/gobuffalo/buffalo/blob/master/buffalo/cmd/test.go#L97-L101

`fm.Up()` uses the previously created test database to run the migrations in case no schema file was found and no development database was present. The generated schema is exported from the test database to the filesystem.

https://github.com/gobuffalo/buffalo/blob/master/buffalo/cmd/test.go#L70

Once the schema is loaded we're back to this line, where it tries to dump the schema into the test database but since the test db was used to generate the schema in the first place it's going to fail.

**Comment by paganotoni** [comment]
@markbates if you have not started on this one i'm willing to help given i've been working recently in gobuffalo-test plugin, if you think i could work on it please LMK :)

**Comment by markbates** [comment]
@paganotoni please! I'm crazy busy. :)

**Comment by paganotoni** [comment]
Solved with #1284
